### PR TITLE
M3-5452: Add support for Autoscaling Node Pools

### PR DESCRIPTION
**Pull request by HanaXu**
## Description
Adds support for Autoscaling a Node Pool on a Kubernetes Cluster's summary page

## How to test
Go to `kubernetes/clusters/<cluser_id>/summary`, ensure the `autoscaler` feature flag is on

- You should see an `Autoscale Pool` button before the Resize Pool button
    - If the Autoscaler is on, you should see min/max values next to the autoscale pool button
    - If the Autoscaler is off, you should not see min/max values next to the autoscale pool button
- Clicking on Autoscale Pool should display a dialog
- The Save Changes button should be disabled until there has been a change in the form and there are no errors
- If Autoscaler is off, the Min/Max input fields should be disabled
- If the Minimum value is not between 1-99 or greater than the Maximum value, there should be an error displayed under the input field
- If the Maximum value is not between 1-100, there should be an error displayed under the input field
- If the Maximum value is less than the current Max, then a warning notice should display
    - Clicking on `resize` in the warning should close the modal and open the resize drawer for that pool
- Clicking Save Changes should make a request to `lke/clusters/<cluster_id>/pools/<pool_id>`
    - Any API errors should be displayed as a Notice error in the dialog
    - A successful request should close the dialog, pop a success toast and update the Min/Max values next to the Autoscale Pool button (might take a few seconds to update)

### CODE REVIEW

**Comment by matthewbartholomew** [comment]
A few small UI adjustments to Pool table and the modal.

**Comment by matthewbartholomew** [comment]
A few small adjustments to the UI. Let me know if you would like to discuss options or challenges. 
Autoscale button and values
Autoscale modal

**Comment by matthewbartholomew** [comment]
**LKE detail page**
(2) If a pool is not autoscaled no min/max value should show
By default—no values
If autoscaling is turned off values are removed from table
(1) Close space between Autoscale Pool button and values to 8px space (-3px)

**Modal**
(3) Open space above toggle (add 12px)
(4) Decrease space above Min/Max fields (-3px)
(5) Make min/max entry fields narrower (70px) (issues with the grid?)
Toggle does not have a label indicating state—that’s a good change from the original designs

File: packages/validation/src/kubernetes.schema.ts [hunk 1]
```diff
@@ -1,10 +1,39 @@
...
 export const nodePoolSchema = object().shape({
   type: string(),
   count: number(),
 });
 
+export const AutoscaleNodePoolSchema = object({
+  enabled: boolean(),
+  min: number().when('enabled', {
+    is: true,
+    then: number().test(
```
Comment (DevDW) [inline packages/validation/src/kubernetes.schema.ts:12]: Not sure if it's necessary, but should `number()` have `.required()` chained onto it here and also for the `max`?

**Review by DevDW — COMMENTED** [review]
Still need to review a couple of more files, but looks good so far. Left a comment/question regarding the schema

**Review by bnussman — COMMENTED** [review]
One thing I noticed is that when I change the autoscaler min / max value on a pool, it takes a few seconds for the `(Min 4 / Max 10)` values beside the button up update. Can we perform an immediate refetch or something to make the experience a bit better?

File: packages/manager/src/features/Kubernetes/KubernetesClusterDetail/NodePoolsDisplay/AutoscalePoolDialog.tsx [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+    onSubmit: submitForm,
+  });
+
+  return (
+    <ConfirmationDialog
+      open={open}
+      title="Autoscale Pool"
+      onClose={handleClose}
+      actions={() => (
+        <form onSubmit={handleSubmit}>
```
Comment (bnussman) [inline packages/manager/src/features/Kubernetes/KubernetesClusterDetail/NodePoolsDisplay/AutoscalePoolDialog.tsx:131]: Is there any benefit to wrapping this in a `<form />` over putting the submit function in the `onClick` of the `<Button />`?

Comment (HanaXu) [reply packages/manager/src/features/Kubernetes/KubernetesClusterDetail/NodePoolsDisplay/AutoscalePoolDialog.tsx:131]: Good point, not that I know of

**Comment by HanaXu** [comment]
> One thing I noticed is that when I change the autoscaler min / max value on a pool, it takes a few seconds for the `(Min 4 / Max 10)` values beside the button up update. Can we perform an immediate refetch or something to make the experience a bit better?

Not sure how to do that 🤔

**Review by tiffwong — COMMENTED** [review]
Everything looks great!

The only question I have is regarding using the touched hook on the inputs to control the disabled state of the Save Changes button. If I make changes to the dialog and reopen it, the button actually stays enabled. 

Also based on the other places where we disable the button if no changes have been detected, I think the button should be disabled based on if the current value matches the initial values instead of depending on if the user has interacted with the input field. 🤔

**Review by DevDW — APPROVED** [review]
LGTM, nice job! 🚀

**Review by tiffwong — APPROVED** [review]
Great job! 🚢

**Review by bnussman — APPROVED** [review]

**Pull request merged**